### ISSUE TITLE
libelektra 0.9.12 regression test failure

### ISSUE THREAD

**Issue by chenrui333** [ElektraInitiative/libelektra#4859]
While trying to [upgrade elektra formula to the latest release](https://github.com/Homebrew/homebrew-core/pull/124723), we found out some regression test failure as below:

```
$ /opt/homebrew/Cellar/elektra/0.9.12/bin/kdb plugin-check spec
There are 1 warnings for this plugin
For high quality plugins there should be no warning
Warning #0: no placements information found
```

relates to https://github.com/Homebrew/homebrew-core/pull/124723

**Comment by markus2330** [comment]
Thank you for reporting this problem! I can reproduce it.

Seems like our build server did not run these tests or the errors were not seen. @tmakar please investigate and fix.

**Comment by atmaxinger** [comment]
Ah yes that's because it's a hook plugin now and as such does not need any placement infos. We should probably add a flag of some kind for these situations.

**Comment by markus2330** [comment]
Yes, the fix should be fairly simple.

The important question is how this could get past our CI and the release process and how to avoid this in future?

**Comment by atmaxinger** [comment]
We do actually have automated tests for these checks in [`check_kdb_internal_check.sh`](https://github.com/ElektraInitiative/libelektra/blob/master/tests/shell/check_kdb_internal_check.sh). However, we whitelist multiple plugins, including the `spec` and `gopts` plugin.

**Comment by tmakar** [comment]
@chenrui333 Does this happen during homebrew installation?

**Comment by tmakar** [comment]
When running `/opt/homebrew/Cellar/elektra/0.9.12/bin/kdb plugin-check spec` I don't get this warning, are there any steps I'm missing?

**Comment by tmakar** [comment]
Ok, my bad, was running with another version.

**Comment by tmakar** [comment]
@atmaxinger @markus2330 

So, I see two solutions to this:
1. Exclude all hook plugins from `plugin-check` in `plugincheck.cpp`
2. Set `infos/placements` to a valid enum value (https://www.libelektra.org/devgettingstarted/contract)

I would prefer option 1, as there is no good reason to place `infos` that are not needed. 

Please let me know what you think!

**Comment by atmaxinger** [comment]
I think we should just check whether the plugin exports something under `system:/elektra/modules/<module name>/exports/hook`. If it does, just assume it's a hook plugin and don't check for other placements.

**Comment by tmakar** [comment]
Ah, ok I see.

Do mean checking it somewhere here:

(Snippet from `plugin.cpp`)
```cpp
if (infos.find ("placements") == infos.end ())
{
   warnings.push_back ("no placements information found");
}
else
{
	std::string placements = infos["placements"];
	if (placements.empty ())
	{
	      warnings.push_back ("placements are empty");
	}

	if (placements == "backend")
	{
		// accepted
		// TODO (flo91): add checks in new mount tooling
	}
```

**Comment by atmaxinger** [comment]
Yes

**Comment by kodebach** [comment]
Why not just add `"hook"` as an accepted value for `infos/placements` like we did with `"backend"` for backend plugins?

**Comment by atmaxinger** [comment]
Yes, this is probably the better idea. We should stay consistent in those cases.

**Comment by tmakar** [comment]
> Why not just add `"hook"` as an accepted value for `infos/placements` like we did with `"backend"` for backend plugins?

Alright!

**Comment by tmakar** [comment]
Ok, it is, but just checked here:

```cpp
if (placements == "backend")
	{
		// accepted
		// TODO (flo91): add checks in new mount tooling
	}
```

Maybe we just mention this here https://www.libelektra.org/devgettingstarted/contract as well.

**Comment by atmaxinger** [comment]
How do we continue with this issue? We have fixed this regression in master. The relevant [patch can be seen here](https://patch-diff.githubusercontent.com/raw/ElektraInitiative/libelektra/pull/4862.patch).

AFAIK homebrew allows applying patches in its formula. So maybe update the formula with this patch?

**Comment by kodebach** [comment]
I think @mpranj should just reopen https://github.com/Homebrew/homebrew-core/pull/124723 with the updated SHA of the fixed version.

**Comment by mpranj** [comment]
The clean way to update the package on homebrew is to release a new tarball. We can't/shouldn't really alter already released tarballs.

So it's either a small re-release or a patch.

Re-release would also bring stable Ubuntu Jammy Jellyfish packages, so not so bad.

**Comment by kodebach** [comment]
In that case I vote for a small 0.9.13 bugfix release.

> Re-release would also bring stable Ubuntu Jammy Jellyfish packages, so not so bad.

We could in theory also add Kinetic Kudu.

**Comment by tmakar** [comment]
@kodebach @mpranj Have we agreed on what we want to do? re-release or patch?

**Comment by mpranj** [comment]
I would agree with previous comments and I vote for a small 0.9.13 bugfix release. In the release notes I would refer to 0.9.12 due to the amount of changes.

> We could in theory also add Kinetic Kudu.

Sure, I'll draft a PR, but this would not be a release blocker imo.

Are there any other votes?

**Comment by mpranj** [comment]
Resolved via #4862.
